public/app/features/explore/Explore.tsx

```diff
@@ -182,10 +183,46 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
     }
   };
 
+  /**
+   * Used by Logs details.
+   */
+  analyzeQuery = async (check: string, key: string, value: string) => {
```

Comment:
> It is, because as modifyQuery originally worked (and still works), it adds the filter (positive or negative) to all the queries, not just the one that produced the log line.

So to keep it consistent maybe it should work as Gabor mentioned in his [comment](https://github.com/grafana/grafana/pull/70328/files#r1247614992):

> the [+] turns blue if every query has the filter: the for loop here returns false as soon as any query answers false, otherwise true.

Reply:
Ohhh I see what you mean, thanks for the reminder because I completely missed this comment.

Reply:
Refactored. Thanks both!

![imagen](https://github.com/grafana/grafana/assets/1069378/45a0f331-003e-42fe-bf8e-8cd7cd8aeb50)

![imagen](https://github.com/grafana/grafana/assets/1069378/2b851b98-cf18-4a67-9027-42c29c2d4131)